### PR TITLE
Imported the generated semantic convention constants

### PR DESCRIPTION
**Pull request by Kwintenvdb**
* Imported the generated semantic convention constants and replaced string literals with constants
* Minor code style improvements (variable names clashing with package names)

### CODE REVIEW

File: core/trace/dt_span_serializer.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ import (
 	"google.golang.org/protobuf/proto"
 
 	"github.com/dynatrace-oss/opentelemetry-exporter-go/core/internal/version"
+	"github.com/dynatrace-oss/opentelemetry-exporter-go/core/semconv"
```
Comment (Oberon00) [inline core/trace/dt_span_serializer.go:26]:
```suggestion
	"github.com/dynatrace-oss/opentelemetry-exporter-go/core/internal/semconv"
```

We should not expose these names to users. Users should just use the public OTel semantic conventions.

Comment (Kwintenvdb) [reply core/trace/dt_span_serializer.go]: Done

File: core/trace/dt_span_serializer.go [hunk 2]
```diff
@@ -254,16 +254,15 @@ func getSerializedResourceForSpanExport(spanResource *resource.Resource) ([]byte
...
-	return proto.Marshal(&resource)
+	return proto.Marshal(&res)
 }
 
 func getExporterResource() *resource.Resource {
 	return resource.NewSchemaless(
-		// TODO get the attribute keys from the generated semantic conventions
-		attribute.Key("telemetry.exporter.name").String("odin"),
-		attribute.Key("telemetry.exporter.version").String(version.FullVersion),
+		attribute.Key(semconv.TelemetryExporterName).String("odin"),
```
Comment (Oberon00) [inline core/trace/dt_span_serializer.go:265]:
```suggestion
		attribute.Key(semconv.TelemetryExporterName).String(semconv.TelemetryExporterNameOdin),
```

There is a constant for that as well.

File: core/trace/dt_span_serializer.go [hunk 3]
```diff
@@ -254,16 +254,15 @@ func getSerializedResourceForSpanExport(spanResource *resource.Resource) ([]byte
 	if err != nil {
 		return nil, err
 	}
-	resource := protoResource.Resource{
```
Comment (Oberon00) [inline core/trace/dt_span_serializer.go:257]: Why this renaming?

**Review by Oberon00 — APPROVED** [review]

File: core/trace/dt_span_serializer.go (hunk 3 above)
Comment (Kwintenvdb) [reply core/trace/dt_span_serializer.go:257]: We already import a package called `resource`. Technically allowed, but bad practice and can lead to confusing compiler errors.

**Comment by Oberon00** [comment]
The inclusive language checker brings up a good point: The links that go to the "master" branch of the Otel spec should be updated to "main" as the branch was renamed.
This would have to be fixed in our spec repository.

**Review by Oberon00 — APPROVED** [review]

**Pull request merged**